api/krusty/localizerunner.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package krusty
+
+import (
+	"sigs.k8s.io/kustomize/api/internal/localizer"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// LocalizeRunner runs `kustomize localize`. Like a CLI subprocess executing the command,
+// it only needs the command arguments and flags and to be injected with a file system to run.
+type LocalizeRunner struct {
+	Scope string
+}
+
+// Run `localize`s referenced files in kustomization target to destination newDir on fSys.
+func (lr *LocalizeRunner) Run(fSys filesys.FileSystem, target string, newDir string) error {
```

Comment:
So I think this is a reasonable way to do the localize tests, but I think we should put this and the localize tests in their own directory if we can, just for organization. Could you put it in `api/krusty/localizetests`? Is everything you need from `krusty` exportable if we take that route?

Reply:
Yes, that works! Done.